lib/kafka/connection.rb

```diff
@@ -89,6 +102,12 @@ def request(api_key, request, response_class)
           end
         end
       end
+    rescue Errno::EPIPE, Errno::ECONNRESET, Errno::ETIMEDOUT => e
+      @logger.error "Connection error: #{e}"
+
+      close
```

Comment:
Unfortunately not – setting up a new connection may take time, so the code responsible for handling retries (Producer) should get to decide whether or not to try to reconnect. You can imagine what would happen if a broker is permanently down: if we tried to reconnect in Connection we'd be stuck in an endless loop.